### ISSUE TITLE
[Feature request] Export to markdown

### ISSUE THREAD

**Issue by quesada** [giuspen/cherrytree#579]

**Comment by Thmyris** [comment]
Any ideas on how this could be implemented? Personally, I don't like the approach taken for exporting to PDF's as it ignores the tree's hierarchical structure and prints every node one under the other with no distinction between nodes and subnodes.

I try to take all my notes in CT, and was bummed out to find out that printing them in PDF form would not be practical for studying. Because every node/subnode was indistinguishable from each other. It would be a mess to go through those papers, understand titles with ease, keep them sorted.

An alternative I could've taken is naming the top nodes in the hierarchy starting with 1. 2. 3... naming their subs a. b. c.... and naming their subs I. II. III.... and so on. 

Another simple solution I've lately been thinking about for expressing node/subnode distinction in markdown would be to use markdown headers. Now that would limit the depth of the tree by the amount of headers you can have in markdown which is six. Depth limitation is another whole can of worms that needs discussion but it would look so nice :)

# Computer Science
xyz
## Data Structures & Algorithms
xyz
### Flowcharts
xyz
#### How to create a flow chart
xyz
#### Benefits of flow charts
xyz
###### Additional Notes on the Benefits
xyz

**Comment by ForeverRainbow** [comment]
@Thmyris I am not really sure how the heirarchy can be maintained when exporting to pdf since it doesnt have a concept of a heirachy, just "flat" pages as it were. You can select the "Node names as headers" and "Each node on seperate page" options to improve the seperation and get a feeling of "ownership" of subnodes as it were. 

Also @quesada I am not sure what this is actually asking for, is it turning cherrytree rich text into markdown on export? Or formatting markdown on export (see #920)? Or something else?

**Comment by Thmyris** [comment]
@ForeverRainbow Yep. I think it could've maybe had different colored and different sized titles for each node. Maybe as you go down the subnodes, the colors would change according to their node cherry color and the title font would become smaller and smaller. 

I believe @quesada means exporting an entire cherrytree document to markdown just like how most other export functions work.

**Comment by phoe** [comment]
+1 vote for the issue.

I think that hierarchicity could be implemented via a directory structure - a series of nested folders, each with an `index.md` file containing the contents of that single node. This way the export would not be a single file, but a whole directory that can then be further processed by other utilities.

**Comment by repsorp** [comment]
I just found this but didn't try it yet
[CherryTree to Markdown converter](https://gitlab.com/kibley/cherrytreetomarkdown)
